### DISCUSSION TITLE
Github for dummys question

### DISCUSSION THREAD

**Discussion by cpottle9** [micropython/micropython#10918] [General]
Hi,

I found a bug in the rp2 port and coded a small fix. 
So now, I want to create a pull request.

I have a great deal of linux based git experience from a previous job.
But, I know nothing about github.

I have my fix coded and tested.
I am building on a Raspberry PI 4 running raspbian.
My local repo is from https://github.com/micropython/micropython.git origin/master.

I googled looking for a step-by-step recipe but I couldn't find anything helpful to me.
Can someone provide a pointer to a resource that describes the procedure.

Thanks!

**Comment by dlech** [comment]
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request

**Reply by dlech** [reply]
Also https://github.com/micropython/micropython/wiki/ContributorGuidelines#sending-pull-requests for extra info specific to MicroPython.

**Comment by peterhinch** [comment]
I also suggest you read [micropython code conventions](https://github.com/micropython/micropython/blob/master/CODECONVENTIONS.md).
